### PR TITLE
fix: prevent non-shared lists from blocking account deletion

### LINKED ISSUE #145: account deletion fails for non-shared lists
Database checks that there is no other admin, should also ensure that there is one other member

### PR DESCRIPTION
**Pull request by Ryangman**
A bug in the database interface to check if a user was the last admin of a list proactively prevented deletion of users who were the only admin of a list that wasn't shared.

This fixes that bug and adds e2e testing to prevent future issues. 

## Related Issues

Closes #145 

## Changes
- [x] fixed bug in database function that checks last admin
- [x] added e2e test to verify behavior
- [x] added cypress command to create dummy test list  

## Testing Coverage
Tested via e2e tests, database action cannot be verified by in vitest

Test coverage issues from deepsource are out of scope for this PR, and will mostly be resolved with the normal vs Github List Modal

### CODE REVIEW

File: cypress/support/commands.ts [hunk 1]
```diff
@@ -73,11 +73,22 @@ Cypress.Commands.add('login', (username: string, password: string) => {
     .and('not.be.null');
 });
 
+Cypress.Commands.add('createList', (listName: string) => {
+  cy.visit('/list');
+
+  cy.get('button .bi-plus').click();
```
Comment (radiantBear) [inline cypress/support/commands.ts]: It would be better to use an aria label or name attribute to find this button rather than a CSS class. If it doesn't have one yet, could you please add that?

File: cypress/support/commands.ts [hunk 2]
```diff
@@ -73,11 +73,22 @@ Cypress.Commands.add('login', (username: string, password: string) => {
...
+  cy.findByPlaceholderText('List name').type(listName);
+  cy.get('button[type="submit"]').click();
+
+  cy.location('pathname').should('match', /^\/list\/[a-zA-Z0-9]{16}$/);
+});
+
```
Comment (radiantBear) [inline cypress/support/commands.ts:85]: More generally, is this a helper function we should even define? It seems to me that it might be better to just expand the seed script to create a few different variations of lists for testing.

**Review by radiantBear — CHANGES_REQUESTED** [review]
Thanks Ryan! That logic tweak makes sense; easy to accidentally overlook that edge case. Thanks for adding an e2e test for it!

File: cypress/support/commands.ts (hunk 2 above)
Comment (Ryangman) [reply cypress/support/commands.ts:85]: I think for these end to end tests we should be using the users flow through the UI as much as possible, for example with my original review on #115, a user couldn't create a list because what the api sent the to the database interface and what the database expected where out of sync. If we manually seeded a list, even e2e tests wouldn't have caught that, but this would.

Comment (radiantBear) [reply cypress/support/commands.ts:85]: Good point! On the other hand, [Cypress' best practices](https://docs.cypress.io/app/core-concepts/best-practices#Organizing-Tests-Logging-In-Controlling-State) note that "taking shortcuts" and "programmatically taking control of application state" are good things. Especially when e2e testing more complex features, like filtering, it seems better to use DB seeding and have a separate test to validate that list creation works.

I think this is fine for now, but we should be careful about how much test-state generation we do by interacting with the site.

Comment (Ryangman) [reply cypress/support/commands.ts:85]: Long term i'd agree, but until we have a test designed to verify list creation, I don't thing we should assume it works by seeding. Id say our best practice would be as soon as we have a test that thoroughly verifies something works we can seed that behavior in other tests, but until that test exists use the user flow. Kind of like how sign up is now.

**Review by radiantBear — DISMISSED** [review]

**Review by sabri414 — DISMISSED** [review]
Looks good!

**Review by radiantBear — APPROVED** [review]

**Review by A-man37 — APPROVED** [review]

**Review by sabri414 — APPROVED** [review]
